tuna/rocmlir/rocmlir_tables.py

```diff
@@ -264,9 +475,146 @@ def set_tables(self, sess_class=SessionRocMLIR):
+        direction = f"{direction} "
+        if "-F" in line:
+          direction = ""
+
+        # Skip filter layout if already in
+        filter_layout = f"-f {layout} "
+        if "-f" in line:
+          filter_layout = ""
+
+        # Skip input layout if already in
+        input_layout = f"-I {layout} "
+        if "-I" in line:
+          input_layout = ""
+
+        # Skip output layout if already in
+        output_layout = f"-O {layout} "
+        if "-O" in line:
+          output_layout = ""
+
+        one_config = f"{datatype}{direction}{filter_layout}\
+                       {input_layout}{output_layout}{line}"
+
+        if one_config not in configs:
+          configs.append(one_config)
+
+    return configs
+
+
+class RocMLIRDBTablesGEMM(RocMLIRDBTables):
+  """Represents db tables for rocMLIR lib when op is conv"""
+
+  def set_tables(self, sess_class=SessionRocMLIR):
+    """Set appropriate tables based on requirements"""
+    super().set_tables(sess_class)
+    # +++pf:  branch on self.config_type when we have GEMM, too.
+    self.job_table = GEMMJob
+    self.config_table = GEMMConfig
+    self.results = GEMMResults
+
+  ## Adapted from perfRunner.getGemmConfigurations.
+
+  DATA_TYPES = ['f32', 'f16', 'i8']
+
+  def get_configurations(self, filename):
+    #pylint: disable=invalid-name
+    """Read gemm-configs from filename and expand into all combinations of
+         type and transpose.
+      """
+
+    configs = []
+    with open(filename, 'r', encoding='utf8') as config_file:
+      lines = config_file.readlines()
+
+      # All combinations of types and transposition (A and B)
+      for datatype, transA, transB, line in \
+              itertools.product(self.DATA_TYPES, ['false', 'true'],
+                                ['false', 'true'], lines):
+        line = line.strip()
+
+        # Skip empty lines
+        if len(line) == 0 or line[0] == '#':
+          continue
+
+        # We need trailing spaces here to account for the concat below
+        # Skip type if already in
+        dataTypeString = ""
+        if "-t " not in line:
+          dataTypeString = f"-t {datatype} "
+
+        # Skip transA if already in
+        transAString = ""
+        if "-transA " not in line:
+          transAString = f"-transA {transA} "
+
+        # Skip transB if already in
+        transBString = ""
+        if "-transB " not in line:
+          transBString = f"-transB {transB} "
+
+        # Skip out_datatype if already in
+        outDataTypeString = ""
+        if "-out_datatype" not in line:
+          outDataTypeString = f"-out_datatype {datatype} "
+
+        # Strip to avoid spurious spaces
+        oneConfig = f"{dataTypeString}{outDataTypeString}\
+                      {transAString}{transBString}{line}".strip()
+        if oneConfig not in configs:
+          configs.append(oneConfig)
+
+        # Special case to get both i8_i8 and i8_i32, w/o --data-type or output-type-map.
```

Comment:
Would generally make i8_i8, unless specified in more detail.  If it's general, add the other case too.